lib/readline.js

```diff
@@ -47,9 +47,7 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
-
-  if (typeof completer !== 'function') {
+  if (completer && typeof completer !== 'function') {
```

Comment:
If no `completer` was explicitly provided, it will be `undefined` here. So valid values at this point are `undefined` (or more precisely, the way the code is, falsy values) or a function. It's only a `TypeError` if a completer was provided *and* it is not a function. So `completer &&`.